### PR TITLE
DOC:  Fix sentence casing in page titles

### PR DESCRIPTION
**Pull request by sayantikabanik**
### PR details
- Fix for the issue - [16261](https://github.com/numpy/numpy/issues/16261)
- Files modified in this PR - `doc/sources` & `doc/neps`
- Files under `doc/release` looks good, no changes required.
### TODO
- The page subtitles also have mixed styles, sentence casing can be introduced
- Docs under `doc/build` needs to be worked upon

### CODE REVIEW

File: doc/neps/nep-0030-duck-array-protocol.rst [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-NEP 30 — Duck Typing for NumPy Arrays - Implementation
+NEP 30 — Duck typing for NumPy arrays - Implementation
 ======================================================
 
 :Author: Peter Andreas Entschev <pentschev@nvidia.com>
 :Author: Stephan Hoyer <shoyer@google.com>
 :Status: Draft
 :Type: Standards Track
-:Created: 2019-07-31
+:Created:
```
Comment (melissawm) [inline doc/neps/nep-0030-duck-array-protocol.rst:11]: Any reason to remove this date?

File: doc/neps/nep-0005-generalized-ufuncs.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .. _NEP05:
 
 =======================================
-NEP 5 — Generalized Universal Functions
+NEP 5 — Generalized universal functions
```
Comment (melissawm) [inline doc/neps/nep-0005-generalized-ufuncs.rst:4]: I have a feeling that Universal Functions is a "proper name" and should be capitalized, but I'd defer to a native English speaker to confirm...

File: doc/neps/nep-0006-newbugtracker.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
 .. _NEP06:
 
 ===================================================
-NEP 6 — Replacing Trac with a different bug tracker
+NEP 6 — Replacing trac with a different bug tracker
```
Comment (melissawm) [inline doc/neps/nep-0006-newbugtracker.rst:4]: Similarly, I think Trac is a proper name and should be capitalized.

File: doc/neps/roadmap.rst [hunk 4]
```diff
@@ -1,5 +1,5 @@
 =============
-NumPy Roadmap
+NumPy roadmap
```
Comment (melissawm) [inline doc/neps/roadmap.rst:2]: I am mixed on this, Roadmap is usually capitalized everywhere I look... I don't have strong feelings either way but not sure about the sentence case rule here.

**Review by melissawm — COMMENTED** [review]
Thank you, @sayantikabanik! I left a few comments where I am still in doubt but because I'm not a native English speaker I might be totally wrong, so feel free to ignore them :)

File: doc/neps/nep-0030-duck-array-protocol.rst (hunk 1 above)
Comment (sayantikabanik) [reply doc/neps/nep-0030-duck-array-protocol.rst:11]: My bad, I shall revert the change.

File: doc/neps/nep-0005-generalized-ufuncs.rst (hunk 2 above)
Comment (sayantikabanik) [reply doc/neps/nep-0005-generalized-ufuncs.rst:4]: I ran a spaCy model to confirm (English - en_core_web_sm (v2.3.0). Its a Proper noun. I shall make  the changes
Screenshot 2021-09-08 at 8 22 57 PM

File: doc/neps/nep-0006-newbugtracker.rst (hunk 3 above)
Comment (sayantikabanik) [reply doc/neps/nep-0006-newbugtracker.rst:4]: I shall make the change. 
Screenshot 2021-09-08 at 8 23 26 PM

File: doc/neps/roadmap.rst (hunk 4 above)
Comment (sayantikabanik) [reply doc/neps/roadmap.rst:2]: Screenshot 2021-09-08 at 8 24 04 PM
It's a proper noun too.

**Comment by melissawm** [comment]
Merging this, any remaining items can be solved in a follow-up. Thanks @sayantikabanik !

**Pull request merged**